gazebo_plugins/src/gazebo_ros_wheel_slip.cpp

```diff
@@ -105,15 +124,123 @@ void GazeboRosWheelSlip::Load(gazebo::physics::ModelPtr _model, sdf::ElementPtr
     };
 
   impl_->on_set_parameters_callback_handle_ = impl_->ros_node_->add_on_set_parameters_callback(
-    param_change_callback);
+    param_validation_callback);
+
+  // Add hook from global slip compliance parameter to the individual wheels
+  auto parameter_set_callback_function = [this](const rcl_interfaces::msg::ParameterEvent & event) {
+      // Only care about this node
+      std::string node_namespace = this->impl_->ros_node_->get_namespace();
+      std::string node_name = this->impl_->ros_node_->get_name();
+      auto complete_node_name = node_namespace + "/" + node_name;
+      if (event.node == complete_node_name) {
+        // Combine the new and changed parameter vectors
+        auto param_list = event.new_parameters;
+        param_list.insert(
+          param_list.end(), event.changed_parameters.begin(),
+          event.changed_parameters.end());
+
+        // Iterate over parameters
+        for (auto & parameter : param_list) {
+          auto temp_param = this->impl_->ros_node_->get_parameter(parameter.name);
+
+          // Propagate global slip compliance lateral parameter to all wheels
+          // This does not set the actual parameter in gazebo
+          if (parameter.name == "slip_compliance_unitless_lateral") {
+            double slip = temp_param.as_double();
+            if (slip >= 0.) {
```

Comment:
I think this is handled by the validation callback and we can remove this case.

Reply:
Done